src/app/core/global/global.service.ts

```diff
@@ -32,44 +36,65 @@ export class GlobalService {
+            this.ipc.request("checkForPlatformUpdates").withLatestFrom(this.localRepository.getIgnoredUpdateVersion())
                 .take(1).subscribe((result) => {
 
-                this.checkingForPlatformUpdate = false;
-                this.statusBar.stopProcess(process, "Checking for platform updates is finished.");
+                this.checkForPlatformUpdatePromise = null;
+                this.statusBar.stopProcess(process, "");
 
-                const [hasUpdate, lastAvailableUpdateStored] = result;
+                const [hasUpdate, ignoredUpdateVersion] = result;
 
                 if (!hasUpdate) {
-                    this.localRepository.setUpdateAvailable(null).then();
-                    return;
-                }
 
-                if (hasUpdate !== lastAvailableUpdateStored) {
-                    const modal = this.modal.fromComponent(UpdatePlatformModalComponent, { title: "Update available!"});
+                    this.localRepository.setIgnoredUpdateVersion(null).then();
+
+                    if (showUpToDateModal) {
+                        const modal = this.modal.fromComponent(UpdatePlatformModalComponent, {title: "Platform updates!"});
+
+                        modal.onCancel = () => {
+                            this.modal.close();
+                        };
+                    }
+
+                } else {
+
+                    this.platformIsOutdated = true;
 
-                    modal.description = hasUpdate.body;
-                    modal.downloadLink = hasUpdate.html_url;
+                    if (hasUpdate !== ignoredUpdateVersion) {
+                        const modal = this.modal.fromComponent(UpdatePlatformModalComponent, {title: "Platform updates!"});
 
-                    modal.onCancel = () => {
-                        this.modal.close();
-                        this.localRepository.setUpdateAvailable(hasUpdate).then();
-                    };
+                        modal.platformIsOutdated = true;
+                        modal.description = hasUpdate.body;
+                        modal.downloadLink = hasUpdate.html_url;
+
+                        modal.onCancel = () => {
+                            this.modal.close();
+                            this.localRepository.setIgnoredUpdateVersion(hasUpdate).then();
+                        };
+                    }
                 }
 
+                resolve(result);
+
             }, (err) => {
-                this.checkingForPlatformUpdate = false;
-                this.statusBar.stopProcess(process, "Failed to get platform updates.");
+                this.checkForPlatformUpdatePromise = null;
+                this.statusBar.stopProcess(process, "");
                 this.notificationBar.showNotification(new ErrorNotification("Cannot get platform updates. " + new ErrorWrapper(err)));
+
+                reject(new ErrorWrapper(err));
```

Comment:
Promise rejection should be given the real error object, not the wrapper. Wrapper should be used just for unifying messages before displaying the error.